src/mvba/vcbc/mod.rs

```diff
@@ -0,0 +1,82 @@
+pub(super) mod context;
+pub(super) mod message;
+pub(super) mod state;
+
+mod deliver;
+mod echo;
+mod error;
+mod propose;
+
+use self::error::Result;
+use self::message::Message;
+use self::propose::ProposeState;
+use self::state::State;
+use super::ProposalChecker;
+use crate::mvba::{broadcaster::Broadcaster, proposal::Proposal};
+use blsttc::{PublicKeySet, PublicKeyShare};
+use std::cell::RefCell;
+use std::rc::Rc;
+
+pub(crate) const MODULE_NAME: &'static str = "vcbc";
+
+// VCBC is a verifiably authenticatedly c-broadcast protocol.
+// Each party $P_i$ c-broadcasts the value that it proposes to all other parties
+// using verifiable authenticated consistent broadcast.
+pub(crate) struct VCBC {
+    ctx: context::Context,
+    state: Box<dyn State>,
+}
+
+impl VCBC {
+    pub fn new(
+        proposer: PublicKeyShare,
```

Comment:
Pass the `index` in instead of the key-share, we can then derive the key-share from `parties` when we need it.

This will allow for much smaller messages since we'll be sending around our index instead of our public key share. When another party wants to validate our signature shares, they will derive the our public_key_share from the public_key_set and verify our signature share against that derived public key.

This saves ~47 bytes per message (and just a good practice when working with threshold crypto)